utils/db.ts

```diff
@@ -1,127 +1,132 @@
+export interface Comment extends InitComment {
+  id: string;
+  createdAt: Date;
+}
 
-    const commentsByUserKey = [
-      "comments_by_users",
-      initComment.userId,
-      commentId,
-    ];
-    const commentsByItemKey = [
-      "comments_by_item",
-      initComment.itemId,
-      commentId,
-    ];
+export async function createComment(initComment: InitComment) {
+  let res = { ok: false };
+  while (!res.ok) {
+    const id = crypto.randomUUID();
+    const commentsByUserKey = ["comments_by_users", initComment.userId, id];
+    const commentsByItemKey = ["comments_by_item", initComment.itemId, id];
+    const comment: Comment = { ...initComment, id, createdAt: new Date() };
 
-    const comment: CommentValue = { ...initComment, createdAt: new Date() };
     res = await kv.atomic()
       .check({ key: commentsByUserKey, versionstamp: null })
       .check({ key: commentsByItemKey, versionstamp: null })
       .set(commentsByUserKey, comment)
       .set(commentsByItemKey, comment)
       .commit();
-  }
-
-  return res;
-}
 
-export interface CommentValue extends InitCommentValue {
-  createdAt: Date;
+    return comment;
+  }
 }
 
-export async function getItemComments(
+export async function getCommentsByItem(
   itemId: string,
   options?: Deno.KvListOptions,
 ) {
-  const iter = await kv.list<CommentValue>({
+  const iter = await kv.list<Comment>({
     prefix: ["comments_by_item", itemId],
   }, options);
   const comments = [];
-  for await (const res of iter) comments.push(res);
+  for await (const res of iter) comments.push(res.value);
   return comments;
 }
 
-export interface InitUserValue {
+interface InitUser {
   id: string;
   stripeCustomerId: string;
 }
 
-export async function createUser(initUser: InitUserValue) {
+export interface User extends InitUser {
+  isSubscribed: boolean;
+}
+
+export async function createUser(initUser: InitUser) {
   const usersKey = ["users", initUser.id];
   const stripeCustomersKey = [
-    "user_ids_by_stripe_customer",
+    "users_by_stripe_customer",
     initUser.stripeCustomerId,
   ];
-
-  const user: UserValue = { ...initUser, isSubscribed: false };
+  const user: User = { ...initUser, isSubscribed: false };
 
   const res = await kv.atomic()
     .check({ key: usersKey, versionstamp: null })
     .check({ key: stripeCustomersKey, versionstamp: null })
     .set(usersKey, user)
-    .set(stripeCustomersKey, user.id)
+    .set(stripeCustomersKey, user)
     .commit();
 
-  if (res === null) {
-    throw new TypeError("User with ID already exists");
+  if (!res.ok) {
+    throw res;
   }
+
+  return user;
 }
 
-export async function getUserIdByStripeCustomerId(stripeCustomerId: string) {
-  const res = await kv.get<UserValue["id"]>([
-    "user_ids_by_stripe_customer",
+export async function getUserById(id: string) {
+  const res = await kv.get<User>(["users", id]);
+  return res.value;
+}
+
+export async function getUserByStripeCustomerId(stripeCustomerId: string) {
```

Comment:
Let's make this function call `kv.get` twice, such as:
- `kv.get("user_id_by_stripe_customer_id", stripeCustomerId)` -> userId
- `kv.get("users", userId)` -> User

Reply:
Done.